src/core/Akka/Pattern/CircuitBreaker.cs

```diff
@@ -226,15 +226,12 @@ public Task WithCircuitBreaker(Func<Task> body)
         /// <param name="body">TBD</param>
         public void WithSyncCircuitBreaker(Action body)
         {
-            var cbTask = WithCircuitBreaker(() => Task.Factory.StartNew(body));
-            if (!cbTask.Wait(CallTimeout))
-            {
-                //throw new TimeoutException( string.Format( "Execution did not complete within the time allotted {0} ms", CallTimeout.TotalMilliseconds ) );
-            }
-            if (cbTask.Exception != null)
-            {
-                ExceptionDispatchInfo.Capture(cbTask.Exception).Throw();
-            }
+            var cts = new CancellationTokenSource(CallTimeout);
+            var task = new Task(body, cts.Token);
+            var cbTask = WithCircuitBreaker(() => task);
+            task.RunSynchronously();
+            if (!task.IsCanceled)
```

Comment:
Yeah this is a much cleaner way of writing what we had before

Reply:
Not only cleaner, before it was racy